tests/functional/contribute/test_signup.py

```diff
@@ -0,0 +1,62 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+from selenium.common.exceptions import TimeoutException
+
+from ...pages.contribute.signup import ContributeSignUpPage
+
+
+@pytest.mark.sanity
+@pytest.mark.nondestructive
+def test_toggle_category_select(base_url, selenium):
+    page = ContributeSignUpPage(base_url, selenium).open()
+    category = page.contributor_categories[0]  # Select coding category
+    area = page.category_areas[0]  # Coding category areas
+    category.select_category
+    assert area.changes_to_required_field
+    assert area.is_displayed
+    category = page.contributor_categories[4]  # Select helping category
+    category.select_category
+    area.wait_to_collapse()
+    assert not area.is_required
+    assert not area.is_displayed
```

Comment:
I think these could either be split into separate tests or we could parameterise them. I also think we could make the test a little more readable by avoiding selecting categories by index. Something like the following could work as a parameterised test:

```python
@pytest.mark.sanity
@pytest.mark.nondestructive
@pytest.mark.parametrize('category, is_required', ['coding', True, 'helping', False])
def test_toggle_category_select(base_url, selenium, category, is_required):
    page = ContributeSignUpPage(base_url, selenium).open()
    page.select_category(category)
    assert page.is_area_displayed
```

This would involve a new method in the page object that clicked a category based on the label, which could be a property of the `ContributorCategory` region. For area, you could return True if any of the areas are displayed instead of matching category to area by index.

After writing out the example above, I think it might make more sense as two separate functional tests.